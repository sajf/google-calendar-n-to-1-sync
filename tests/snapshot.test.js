// tests/snapshot.test.js
/* eslint-env jest, node */
require('./mocks');
const { createMockEvent } = require('./mocks');

describe('Snapshot Tests', () => {
  describe('Event Data Structures', () => {
    it('should match event payload snapshot', () => {
      const sourceEvent = {
        id: 'event123',
        summary: 'Test Event',
        description: 'Test Description',
        location: 'Test Location',
        start: { dateTime: '2023-01-01T10:00:00Z' },
        end: { dateTime: '2023-01-01T11:00:00Z' },
        attendees: [
          { email: 'user1@test.com', responseStatus: 'accepted' },
          { email: 'user2@test.com', responseStatus: 'needsAction' }
        ],
        reminders: {
          useDefault: false,
          overrides: [
            { method: 'email', minutes: 15 },
            { method: 'popup', minutes: 10 }
          ]
        }
      };

      // This would create a snapshot of the event structure
      expect(sourceEvent).toMatchSnapshot();
    });

    it('should match sync event payload snapshot', () => {
      const syncEvent = createMockEvent({
        id: 'sync123',
        summary: 'Synced Event',
        extendedProperties: {
          private: {
            SYNC_KEY: 'source@cal.com:original123',
            SYNC_SOURCE: 'source@cal.com',
            SYNC_ORIGINAL_ID: 'original123'
          }
        }
      });

      expect(syncEvent).toMatchSnapshot();
    });

    it('should match complex event map snapshot', () => {
      const events = [
        createMockEvent({
          id: 'event1',
          summary: 'Event 1',
          extendedProperties: {
            private: {
              SYNC_KEY: 'source1:original1',
              SYNC_SOURCE: 'source1'
            }
          }
        }),
        createMockEvent({
          id: 'event2',
          summary: 'Event 2',
          extendedProperties: {
            private: {
              SYNC_KEY: 'source2:original2',
              SYNC_SOURCE: 'source2'
            }
          }
        })
      ];

      expect(events).toMatchSnapshot();
    });
  });
});
