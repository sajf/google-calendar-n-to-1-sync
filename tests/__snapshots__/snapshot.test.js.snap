// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Tests Event Data Structures should match complex event map snapshot 1`] = `
[
  {
    "extendedProperties": {
      "private": {
        "SYNC_KEY": "source1:original1",
        "SYNC_SOURCE": "source1",
      },
    },
    "id": "event1",
    "status": "confirmed",
    "summary": "Event 1",
    "updated": "2023-01-01T00:00:00.000Z",
  },
  {
    "extendedProperties": {
      "private": {
        "SYNC_KEY": "source2:original2",
        "SYNC_SOURCE": "source2",
      },
    },
    "id": "event2",
    "status": "confirmed",
    "summary": "Event 2",
    "updated": "2023-01-01T00:00:00.000Z",
  },
]
`;

exports[`Snapshot Tests Event Data Structures should match event payload snapshot 1`] = `
{
  "attendees": [
    {
      "email": "user1@test.com",
      "responseStatus": "accepted",
    },
    {
      "email": "user2@test.com",
      "responseStatus": "needsAction",
    },
  ],
  "description": "Test Description",
  "end": {
    "dateTime": "2023-01-01T11:00:00Z",
  },
  "id": "event123",
  "location": "Test Location",
  "reminders": {
    "overrides": [
      {
        "method": "email",
        "minutes": 15,
      },
      {
        "method": "popup",
        "minutes": 10,
      },
    ],
    "useDefault": false,
  },
  "start": {
    "dateTime": "2023-01-01T10:00:00Z",
  },
  "summary": "Test Event",
}
`;

exports[`Snapshot Tests Event Data Structures should match sync event payload snapshot 1`] = `
{
  "extendedProperties": {
    "private": {
      "SYNC_KEY": "source@cal.com:original123",
      "SYNC_ORIGINAL_ID": "original123",
      "SYNC_SOURCE": "source@cal.com",
    },
  },
  "id": "sync123",
  "status": "confirmed",
  "summary": "Synced Event",
  "updated": "2023-01-01T00:00:00.000Z",
}
`;
